### DISCUSSION TITLE
Configure  AKS Clusters backup completely on On-Prem Data center

### DISCUSSION THREAD

**Discussion by NagaTeja12** [vmware-tanzu/velero#6782] [Community support Q&A]
Hi All , 

We are asked to set up AKS (Azure) backup completely on On-Prem Data Center using Velero. Please suggest the best practices and things required to set this up .
I have tried different Options but none are useful as the request here is to have the Backup on DC without involving any of the Cloud resources. Any leads or existing setup to have as the reference would be really appreciated. 

Thanks, 
Naga.

**Comment by draghuram** [comment]
Hi @NagaTeja12, Considering that you want to backup AKS, you are going to use cloud, at least for source. By on-prem, do you mean that backup target would be on-prem? A clearer description of requirements would make it easy to suggest solutions.

**Reply by NagaTeja12** [reply]
Hi @draghuram , Sorry for the confusion 
The source is AKS (Azure Cloud) on which we have all the deployments and we are planning to backup it on our On-Prem DC.
The requirement is that if anything goes wrong on Azure end , we need to have our backup in our data Center so that we can deploy them on any other cloud vendors .Hope this is clear and looking for a possible solution.

**Comment by draghuram** [comment]
Thanks, it is sort of clear now. 

If you are planning to use Velero, you need to have access to object storage that is present in your DC (from AKS cluster set up with something like Azure private link). Once such access is there, you can use Velero to do file system backups (or new 1.12's snapshot data backup). You can then restore to any target cluster from your on-prem backups as long as the object storage is accessible from the target cluster.  You do need to map storage classes and fix ingress/load balancer configuration. 

If you are interested in discussing this in detail, you are free to contact me (raghu at catalogicsoftware). 

Thanks, Raghu (CloudCasa).

**Reply by NagaTeja12** [reply]
Thank You Very Much for the info , Will definitely reach out to you once we have everything in place.

**Reply by draghuram** [reply]
Great, thanks.
